src/borg/testsuite/repository.py

```diff
@@ -133,6 +133,7 @@ def test_single_kind_transactions(self):
     def test_list(self):
         for x in range(100):
             self.repository.put(H(x), b'SOMEDATA')
+        self.repository.commit()
```

Comment:
I suspect this was needed to close the segment file that was written to in the line above.
Without the commit, the index-based list() worked, but not the segment-file-based.